typings/index.d.ts

```diff
@@ -1280,16 +1285,42 @@ declare module 'discord.js' {
 
 	export class WebSocketShard extends EventEmitter {
 		constructor(manager: WebSocketManager, id: number);
+		private sequence: number;
+		private sessionID?: string;
+		private lastPingTimestamp: number;
+		private lastHeartbeatAcked: boolean;
+		private trace: string[];
+		private ratelimit: { queue: Object, total: number, reamining: number, time: number, timer: NodeJS.Timer | null };
+		private connection: WebSocket | null;
+		private inflate: Inflate | null;
+		private sendHeartbeat(): void;
+		private setHeartbeatTimer(time: number): void;
+		private ackHeartbeat(): void;
+		private connect(): void;
+		private onOpen(): void;
+		private onMessage(event: Event): void;
+		private onPacket(packet: Object): void;
+		private identify(): void;
+		private identifyNew(): void;
+		private identifyResume(): void;
+		private onError(error: Error): void;
+		private onClose(event: CloseEvent): void;
+		private send(data: Object): void;
```

Comment:
Though marked as private in the docs, we intentionally left this public for extensions. Not sure if we still want to keep this public in the typings or make it public in the docs altogether.

Reply:
I have marked them as `public` for now.